pilot/pkg/bootstrap/istio_ca.go

```diff
@@ -128,8 +131,9 @@ var (
 		"Specify the RSA key size to use for self-signed Istio CA certificates.")
 
 	// TODO: Likely to be removed and added to mesh config
+	// DEPRECATED - Only used to enable k8s signing - presence of K8S_SIGNER used instead to indicate use of external CA.
 	externalCaType = env.Register("EXTERNAL_CA", "",
-		"External CA Integration Type. Permitted value is ISTIOD_RA_KUBERNETES_API.").Get()
+		"DEPRECATED - External CA Integration Type. Permitted value is ISTIOD_RA_KUBERNETES_API.").Get()
```

Comment:
Did the community agree to deprecate this?

Reply:
So far the 'agree to deprecated' feature flags is by approving the PR. We had quite a few removals - I don't think we had any formal process or TOC ( it is not a v1 API - there is already a comment that 'it may be removed and moved to MeshConfig).

If you have a use case for requiring 2 env variables to be set to enable the RA - happy to remove the comment.

Reply:
To clarify: enabling the K8S RA involves setting the K8S_SIGNER to a non-empty value AND setting EXTERNAL_CA to the only value supported, ISTIOD_RA_KUBERNETES_API. 

"Deprecating" means that in the next version we can enable by only setting K8S_SIGNER. 

There is no scenario where one will be used without the other.

Reply:
Are we ok with locking ourselves in with only Kubernetes as the CA? I don't intend for that to sound as leading as it probably does, but I want to confirm we're ok with that

Reply:
Would be good if we are going to deprecate something to have a smaller PR. This PR is a big pile of fixing comments, refactorings, major changes, and deprecation. This makes it very hard to review.

Changing this particular seems fairly reasonable me because we don't need 2 flags to enable one thing, but its hard to discuss in the context of this really big PR fixing a lot of things.

TBH in other areas of the code it probably wouldn't be so bad, but not (m)any are experts so its making it troublesome

Reply:
I can remove this deprecation, it's just text indicating it will be removed, I don't remove the second flag in this PR.

The comments should not have any impact ( they helped me understand what is going on, because the code is extremely confusing, and having spaghetti in security is a risk). No problem with removing them and not merging the PR, now I understand a bit the code.

The only real fix is in the istio_ca and handling of tls.crt ( split leaf from intermediates ), to fix ztunnel - I'm going in vacation so someone will need to pick that, and I can close this PR.